src/includes/traits/Plugin/OptionUtils.php

```diff
@@ -10,9 +10,13 @@
      *
      * @since 151002 Improving multisite compat.
      *
+     * @param bool  $intersect Discard options not present in $this->default_options
+     *
      * @return array Plugin options.
+     *
+     * @note $intersect should be `false` when this method is called via a VS upgrade routine or during inital startup on when upgrading. See https://git.io/viGIK
      */
-    public function getOptions()
+    public function getOptions($intersect = true)
     {
         if (!($options = $this->options)) { // Not defined yet?
```

Comment:
Let me know if that's not clear. I can open a new fork and try to explain in code.